src/ReactiveUI.Winforms/ActivationForViewFetcher.cs

```diff
@@ -44,12 +46,40 @@ public IObservable<bool> GetActivationForView(IActivatable view)
                 }
 
                 return controlActivation;
-            }
-            else {
+            } else {
                 // Show a friendly warning in the log that this view will never be activated
                 this.Log().Warn("Expected a view of type System.Windows.Forms.Control but it is {0}.\r\nYou need to implement your own IActivationForViewFetcher for {0}.", view.GetType());
                 return Observable<bool>.Empty;
             }
         }
+
+        private bool GetCachedIsDesignMode(Control control)
+        {
+            if (_isDesignModeCache == null) {
+                _isDesignModeCache = new Lazy<bool>(() => {
```

Comment:
Just create a constructor and put this in there (without the null check). 

There is a performance penalty for the if() check that is unnecessary.

Also you can simplify the statement and turn off multi threaded protection by passing null to the second parameter

```
_isDesignModeCache = new Lazy>bool>(() => GetIsDesignModel(control), false);
```

Reply:
The control is not available there. How should that work in the constructor?